src/Microsoft.SqlTools.ServiceLayer/Connection/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -806,6 +817,23 @@ public static ServerInfo GetServerVersion(IDbConnection connection)
                     serverInfo.OsVersion = reader[0].ToString();
                 });
 
+                int? isBigDataCluster = serverInfo.Options[ServerInfo.OptionIsBigDataCluster] as int?;
+                if (isBigDataCluster != null && isBigDataCluster.Value == 1)
```

Comment:
I like this implementation, where we add the 1 extra call if needed but in most cases we're not affected. This seems like a good tradeoff.